src/include/constants_general.php

```diff
@@ -169,6 +169,8 @@
 define('SYSLOG_NOTICE', 5);
 define('SYSLOG_INFO', 6);
 define('SYSLOG_DEBUG', 7);
+define('SYSLOG_TRACE', 8);
```

Comment:
This priority is actually impossible to map because priority is a 3 Bit and facility a 5 bit Number both combined in a one 8 Bit number. So I would like the SYSLOG_TRACE modification bneing removed from this PR.

Reply:
Ok, i'll try to figure it out how to remove it without affecting my own branch. Maybe i'll need to create a separate branch...
Most of the suggestions i've fixed, yet right now this one is a dealbreaker.

Reply:
Just out of curiosity: why do you need non-existing SYSLOG_TRACE? Any non-syslog sources inside the database?

Reply:
Well, I route logs from my router (Mikrotik Chateau) to rsyslog and view them using loganalizer.
RouterOS have a lot of  [topics](https://wiki.mikrotik.com/wiki/Manual:System/Log) instead of standard log levels, which is cool, but needs special handling. Of course, it can format logs in bsd compliant format (as defined in RFC 3164), but it's not informative enough. 
So, there are topics like "packet, raw" and maybe some other very verbose, that floods logs very fast if enabled. And there is "debug" topic (and maybe some other) that is less verbose. Obviously, I want to separate them out, and here comes in TRACE level. Please, see [Severity Occurences chart](https://imgur.com/vJe6CNy.png).

To mark needed log messages as TRACE i use rsyslog useful filters. 

Moreover, i believe, loganalizer should support such levels, at least, because it supports apache logs which in turn have [trace log level](https://drill.apache.org/docs/logging-and-tracing/). But, IMHO, supported levels should be configurable for each source separately (because rsyslog allows to override real values). 

By the way, these tools are awesome, a huge thanks to you and your team for developing them and making open source!

Reply:
Ok I understand what you mean. Technically I only have a problem mapping the TRACE level as Syslog Priority which is not possible from a protocol point of view. But of course I understand that you can put more values into a database syslog source for the syslog severity.

If you could renamed SYSLOG_TRACE into "EXTENDED_TRACE" for example (open for other suggestions) and give it a number above 255, it would be clearly separated from Syslog Priorities (0-7) and all your filtering enhancements would still work.

Reply:
I suppose "CUSTOM_TRACE"  will be good for this purpose (or maybe "SEVERITY_TRACE"). 
However, do we really need to change its number to above 255? I don't think that syslog priority matrix will be changed in the near future or ever. So, no syslog severity level will ever collide with trace level mapped in loganalyzer. Is LogAnalyzer somewhere recalculating priority or severity values from logs?

Please, don't understand me wrong, I'm not against that, I just don't get how this number affects loganalyzer. If LogAnalyzer is not doing any math (at least i haven't found that in code, please, point me if there is), but using values as they are provided, i see no problem here. As far as i understand, LogAnalyzer is a journal viewer, not recorder or even redactor. (that's why these levels shouldn't be constants, but configurable)
Please, elaborate your point a bit more in this context.

Reply:
The Syslog Severity is stored as 32bit signed integer in the database, so any number from -2147483648 to +2147483648 would be valid. Anyway I would be fine with CUSTOM_TRACE as it clearly separates it from the constant syslog priorities. 

And yes technically it is vey unlikely that those values will every be changed, so I am ok with using 8.